apps/api-extractor/src/collector/Collector.ts

```diff
@@ -196,12 +198,25 @@ export class Collector {
       this.messageRouter.logDiagnosticFooter();
 
       this.messageRouter.logDiagnosticHeader('Files analyzed by compiler');
-      for (const sourceFile of this.program.getSourceFiles()) {
+      for (const sourceFile of sourceFiles) {
         this.messageRouter.logDiagnostic(sourceFile.fileName);
       }
       this.messageRouter.logDiagnosticFooter();
     }
 
+    // We can throw this error earlier in CompilerState.ts, but intentionally wait until after we've logged the
+    // associated diagnostic message above to make debugging easier for developers.
+    const badSourceFile: boolean = sourceFiles.some(
+      ({ fileName }) => !ExtractorConfig.hasDtsFileExtension(fileName)
+    );
+    if (badSourceFile) {
+      throw new Error(
```

Comment:
Maybe we should include the filename in the error message.

Reply:
There could be multiple bad source files. Instead of listing all of them, the error message directs the user to "Run with the "--diagnostics" flag and inspect the "Files analyzed by compiler" to find the unexpected file(s)".

Reply:
How about this? https://github.com/zelliott/rushstack/pull/1

Reply:
LGTM